docs/intro/contributing.txt

```diff
@@ -103,26 +120,96 @@ Download the Django source code repository using the following command:
 
     $ git clone https://github.com/django/django.git
 
-.. note::
+Now that you have a local copy of Django, you can install it just like you would
+install any package using ``pip``. The most convenient way to do so is by using
+a *virtual environment* (or virtualenv) which is a feature built into Python
+that allows you to keep a separate directory of installed packages for each of
+your projects so that they don't interfere with each other.
+
+It's a good idea to keep all your virtualenvs in one place, for example in
+``.virtualenvs/`` in your home directory. Create it if it doesn't exist yet:
+
+.. code-block:: console
+
+    $ mkdir ~/.virtualenvs
 
-    For users who wish to use `virtualenv`__, you can use:
+Now create a new virtualenv by running:
+
+.. code-block:: console
+
+    $ python3 -m venv ~/.virtualenvs/djangodev
+
+The path is where the new environment will be saved on your computer.
+
+.. admonition:: For Windows users
+
+    Using the built-in ``venv`` module will not work if you are also using the
+    Git Bash shell on Windows, since activation scripts are only created for the
+    system shell (``.bat``) and PowerShell (``.ps1``). Use the ``virtualenv``
+    package instead:
+
+    .. code-block:: none
+
+        $ pip install virtualenv
+        $ virtualenv ~/.virtualenvs/djangodev
+
+.. admonition:: For Ubuntu users
+
+    On some versions of Ubuntu the above command might fail. Use the
+    ``virtualenv`` package instead, first making sure you have ``pip3``:
 
     .. code-block:: console
 
-        $ pip install -e /path/to/your/local/clone/django/
+        $ sudo apt-get install python3-pip
+        $ pip3 install virtualenv
```

Comment:
I'm a bit surprised sudo isn't required (it didn't seem to be required on 15.04 as packages are installed in ~/.local), did you test on a fresh 14.04 install). Maybe just add a comment like  `# prefix this command with sudo if it gives a permission denied error`